File: smartele/src/services/WishlistService.js
import httpAxios from "../httpAxios";


const WishlistService = {
    addToWishlist:(userId, productId, qty, price, image) => {
      return httpAxios.post(`/wishlist/add/${userId}/${productId}/${qty}/${price}/${image}`)
        .then(() => {
          console.log("Product added to cart successfully.");
        })
        .catch((error) => {
          console.error('Error adding to cart:', error);
          throw error;
        });
    }, 
 removeFromWishlist:(productId) => {
  return httpAxios.delete( `/wishlist/remove/${productId}`);
},

getAll:() =>{
  return httpAxios.get(`/wishlist/items`);
}

}

export default WishlistService;

File: api (1)/api/api/src/main/java/com/example/api/service/WishlistService.java
package com.example.api.service;

import java.util.List;


import com.example.api.entity.Wishlist; 

public interface WishlistService { 
    void addItemToWishlist(Long userId, Long productId, int qty, double price,String image);
    void removeItemFromWishlist(Long productId);
    List<Wishlist> getWishlistItems();
}
